lib/plugin.js

```diff
@@ -75,7 +76,12 @@ ManifestPlugin.prototype.apply = function(compiler) {
     // This allows output path to be reflected in the manifest.
     if (this.opts.basePath) {
       manifest = _.reduce(manifest, function(memo, value, key) {
-        memo[this.opts.basePath + key] = this.opts.basePath + value;
+        // only if useBasePathAndPublicPath is set, both basePath and publicPath are used
+        if (this.opts.useBasePathAndPublicPath && this.opts.publicPath) {
```

Comment:
you could do this check to build the callback instead of checking it inside the callback for each step of the reducing. This condition does not depend on the step